server/services/auth.js

```diff
@@ -0,0 +1,52 @@
+import jwt from 'jsonwebtoken';
+
+import config from '../../config/env';
+import User from '../models/user';
+import userService from '../services/user';
+
+function loginUser(email) {
+  return new Promise((resolve) => {
+    User.getByEmail(email)
+      .then(() => {
+        const token = jwt.sign({ email }, config.auth.jwt_secret);
+        resolve(token);
+      })
+      .catch((err) => {
+        resolve(err);
+      });
+  });
+}
+
+function signupUser(body) {
+  const userObj = {
+    email: body.email,
+    name: body.name
+  };
+
+  const returnObj = {};
+
+  return new Promise((resolve) => {
+    User.getByEmail(userObj.email)
+      .then(() => {
+        returnObj.error = 'User already exists';
+        resolve(returnObj);
+      })
+      .catch(() => {
```

Comment:
I don't really like having this logic on the `.catch()` as it is right now. Because if something actually errors, and it's not just a DB call that returned a user, then we are creating a user in that case as well. 

I think a separate call that checks to see if an email exists in the DB would be better here.